### PR TITLE
Improve weighted voting

### PR DESCRIPTION
**Pull request by selankon**
- [x] Implements a voting power component
- [x] It show if its weighted on the election type badge

It helps to fix https://github.com/vocdoni/ui-scaffold/issues/532 and blocks https://github.com/vocdoni/ui-scaffold/pull/669

### CODE REVIEW

File: packages/chakra-components/src/components/Election/Questions.tsx [hunk 1]
```diff
@@ -410,10 +415,14 @@ const QuestionsTypeBadge = () => {
...
   }
 
+  const weighted = Number(election.census.weight) !== election.census.size && localize('question_types.weighted_voting')
+
   return (
     <chakra.div __css={styles.box}>
       <Tooltip label={tooltip} hasArrow sx={styles.tooltip} placement='auto'>
-        <chakra.label __css={styles.title}>{title}</chakra.label>
+        <chakra.label __css={styles.title}>
+          {title} {weighted}
```
Comment (elboletaire) [inline packages/chakra-components/src/components/Election/Questions.tsx:424]: Concatenated strings should not be used on a localized application. Use a translation with interpolation instead.

File: packages/chakra-components/src/components/Election/VoteButton.tsx [hunk 2]
```diff
@@ -67,3 +68,26 @@ export const VoteButton = (props: ButtonProps) => {
 
   return <Button {...button} />
 }
+
+export const VoteWeight = () => {
+  const { client, election, localize } = useElection()
+  const [weight, setWeight] = useState<string | null>(null)
+  const styles = useMultiStyleConfig('VoteWeight')
+
+  useEffect(() => {
```
Comment (elboletaire) [inline packages/chakra-components/src/components/Election/VoteButton.tsx:77]: Please always try to add comments to effects, explaining what the effect does.

File: packages/chakra-components/src/components/Election/VoteButton.tsx [hunk 3]
```diff
@@ -67,3 +68,26 @@ export const VoteButton = (props: ButtonProps) => {
...
+    ;(async () => {
+      if (!client || !election || !client.wallet || !election.census.censusId) return
+      const proof = await client.fetchProof(election.census.censusId, await client.wallet.getAddress())
+      setWeight(proof.weight)
```
Comment (elboletaire) [inline packages/chakra-components/src/components/Election/VoteButton.tsx:81]: Add this on a try/catch.

File: packages/chakra-components/src/theme/vote.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      mt: 4,
+      w: 'full',
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      gap: 2,
+    },
+    weight: {
+      fontWeight: 'bold',
+    },
```
Comment (elboletaire) [inline packages/chakra-components/src/theme/vote.ts:25]: Way too much styling for the base component IMHO

**Review by elboletaire — CHANGES_REQUESTED** [review]

File: packages/chakra-components/src/components/Election/Questions.tsx (hunk 1 above)
Comment (selankon) [reply packages/chakra-components/src/components/Election/Questions.tsx]: 6ddfcbf fix it. 

But... @jpaulet could not be best if I just add a new "badge" for weighted voting? On this way the weighted voting badge can have its own tooltip and we can avoid to adapt every location for the badges with the weighted parameter (see the commit)

File: packages/chakra-components/src/components/Election/VoteButton.tsx (hunk 3 above)
Comment (selankon) [reply packages/chakra-components/src/components/Election/VoteButton.tsx]: 8c4a788

File: packages/chakra-components/src/components/Election/VoteButton.tsx (hunk 2 above)
Comment (selankon) [reply packages/chakra-components/src/components/Election/VoteButton.tsx:79]: 8c4a788

File: packages/chakra-components/src/theme/vote.ts (hunk 4 above)
Comment (selankon) [reply packages/chakra-components/src/theme/vote.ts:25]: Is not to much, just for avoid to style on the ui-scaffold.

Comment (selankon) [reply packages/chakra-components/src/theme/vote.ts]: This bf4c190 is simplier and doesn't breaks nothing on ui scaffold. I just added a margin bottom there: [f71f4b6](https://github.com/vocdoni/ui-scaffold/pull/669/commits/f71f4b6)

**Review by elboletaire — APPROVED** [review]

**Pull request merged**